core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -546,6 +546,14 @@ public void start(Boolean isPrimary) throws IOException {
         }
       }
 
+      // File System Master and Block Master Integrity Check
+      if (Configuration.getBoolean(PropertyKey.MASTER_STARTUP_BLOCK_CHECK_ENABLED)) {
+        mBlockMaster.validateBlocks((blockId) -> {
```

Comment:
optional: if we want to avoid the callback, we could have the block master expose a list of its blocks, then have the filesystem master use that list + the `removeBlocks` API to do the integrity check